src/m365/spo/commands/group/group-user-add.ts

```diff
@@ -0,0 +1,111 @@
+import { Logger } from '../../../../cli';
+import {
+  CommandError,
+  CommandOption
+} from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import SpoCommand from '../../../base/SpoCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  webUrl: string;
+  groupId: number;
+  userName: string;
+}
+
+class SpoGroupUserAddCommand extends SpoCommand {
+
+  public get name(): string {
+    return commands.GROUP_USER_ADD;
+  }
+
+  public get description(): string {
+    return 'Adds user to a SharePoint Group';
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: (err?: any) => void): void {
+
+    if (this.verbose) {
+      logger.logToStderr(`Start adding user/s to SharePoint Group ${args.options.groupId}...`);
+    }
+
+    const data: any = {
+      url: `${args.options.webUrl}`,
+      peoplePickerInput: this.getFormatteduserList(args.options.userName),
+      roleValue: `group:${args.options.groupId}`
+    };
+
+    const requestOptions: any = {
+      url: `${args.options.webUrl}/_api/SP.Web.ShareObject`,
+      headers: {
+        'Accept': 'application/json;odata=nometadata',
+        'Content-type': 'application/json;odata=verbose'
+      },
+      data: data,
+      responseType: 'json'
+    };
+
+    request.post<string>(requestOptions)
+      .then((rawRes: any): void => {
+        if (rawRes.StatusCode !== 0) {
+          return cb(new CommandError(rawRes.ErrorMessage));
+        }
+        logger.log(rawRes.UsersAddedToGroup);
```

Comment:
If you specify a non-existent user name, the command fails with an unhandled exception because the `rawRes.UsersAddedToGroup` property is null. We should check for this in our code.

Reply:
That is weird. When I am executing the username which does not exist, I am getting "Couldn't resolve user" in the errormessage property. 🤔.
![image](https://user-images.githubusercontent.com/10369964/123596378-c10e9700-d80f-11eb-8921-b7723e9db327.png)

Not sure whether I am overlooking anything else in the code.

Reply:
Thinking of it, I wonder if it wouldn't be better to check first if the specified user exists, to avoid creating unnecessary invites that will clutter the UI. What do you think @arjunumenon?

Reply:
Good point when you consider the links getting cluttered. Validation check make sense to avoid that situation.

Just to understand, What is in your mind on how to handle that. Do you prefer to,

1. Have a validation check before the command gets executed. So if there are any users who do not exist, command will not be executed and validation fails
2. Or when you execute the command, we would only consider the users which exists currently.

For usability point of view, **No 2** would be preferable. But do you think it would burden the command execution? When we have multiple users in the list of users, we may have to check each users during the execution phase and I was wondering it may impact the execution time. Pardon me, just playing Devil's Advocate here. 😈. 

Your thoughts please @waldekmastykarz ?

Reply:
Unfortunately, right now we can't execute a validation request in the validation stage because then the auth hasn't been restored yet. So the only option we have available is to perform the necessary checks while executing the command.

Reply:
Thanks Waldek. In that case, what I was thinking is to have the check before executing the command and pass only the active / resolved users to the `Sharing` Method. Hope that approach works.

Another question on the same line. Since we already have a CLI command which gets the username, 

- Can we call another command from one command. Is that a recommended approach?
- Or do you prefer to make another REST call which checks the user?

Reply:
We can totally call another command. If it does exactly what you need, it would help us centralize the code. To do that, you'd use the `Cli.executeCommandWithOutput` method: https://github.com/pnp/cli-microsoft365/blob/04b66058b775750239a2b18dbc46634eba86257d/src/cli/Cli.ts#L233

Reply:
That is awesome. Had not tried that before. Let me try that and will see if it works as expected. Should work. Will update the code accordingly.